dora/core/common/src/main/java/alluxio/membership/AlluxioEtcdClient.java

```diff
@@ -107,8 +108,15 @@ public AlluxioEtcdClient(AlluxioConfiguration conf) {
         String.format("%s%s%s", BASE_PATH, MembershipManager.PATH_SEPARATOR, clusterName));
     // TODO(lucy) add more options as needed for io.etcd.jetcd.ClientBuilder
     // to control underneath grpc parameters.
-    mClient = Client.builder().endpoints(mEndpoints)
-        .build();
+    ClientBuilder jetcdClientBuilder = Client.builder().endpoints(mEndpoints);
+    if (conf.isSet(PropertyKey.ETCD_USERNAME) && conf.isSet(PropertyKey.ETCD_PASSWORD)) {
+      jetcdClientBuilder
+          .user(ByteSequence.from(
+              conf.getString(PropertyKey.ETCD_USERNAME), StandardCharsets.UTF_8))
```

Comment:
nit: we only connect with username/pwd if both are specified. I feel it's helpful to leave a trace if only one of them is specified. So by human error only one is set and the user may wonder why isn't the connection using auth (and gets rejected). We can print a warning or sth.

Also we can print an INFO log like `LOG.info("Connecting to ETCD with username {}")`

Reply:
done, add precondition check